src/terminal/init.lua

```diff
@@ -340,17 +353,21 @@ end
 -- @return true
 -- @within cursor_position
 function M.cursor_set(row, column)
-  output.write(M.cursor_sets(row, column))
+  -- Retrieve terminal dimensions once
+  local size = sys.get_terminal_size()
+  row = utils.resolve_index(row, size.height)
+  column = utils.resolve_index(column, size.width)
+
+  -- Write the cursor position to the terminal
+  output.write("\27[" .. tostring(row) .. ";" .. tostring(column) .. "H")
```

Comment:
why this change? if the function `cursor_sets` already does the negative-index-update, then we don't need to change this function at all right?